src/chfn.c

```diff
@@ -648,11 +648,19 @@ int main (int argc, char **argv)
 	 * name, or the name getlogin() returns.
 	 */
 	if (optind < argc) {
-		user = argv[optind];
+		if (is_valid_user_name (argv[optind])) {
+			user = xstrdup (argv[optind]);
+		}
+		else {
+			fprintf (stderr, _("%s: Provided user name is not a valid name\n"), Prog);
+			fail_exit (E_NOPERM);
+		}
+
 		pw = xgetpwnam (user);
 		if (NULL == pw) {
 			fprintf (stderr, _("%s: user '%s' does not exist\n"), Prog,
 			         user);
+			free (user);
```

Comment:
We don't need to free(3) in these early exit() points.

Reply:
Fixed

Reply:
Not really.  I still see it there.

Reply:
You are correct, just the memory was already allocated, so I added free before exiting. Why we should not do free at early exit stage?

Reply:
If we do not do free for any allocated memory, we get issues in other code analyses tools (for example using gcc compiler with allocation checks flags).

Reply:
> You are correct, just the memory was already allocated, so I added free before exiting. Why we should not do free at early exit stage?

Because exit(3) makes sure all memory is released.  And it's just simpler (no need for having dozens of free(3) calls on the same resource).

Reply:
> If we do not do free for any allocated memory, we get issues in other code analyses tools (for example using gcc compiler with allocation checks flags).

Please show those warnings.

Reply:
Some comment to exit(3):
While it is true that exit(3) ensures that all memory is freed, it is still considered good practice to explicitly free dynamically allocated memory using free() before calling exit(3) or before the program exits normally. This practice has several benefits:
-     Portability: On some non-Linux systems, exit(3) might not guarantee that all resources are properly cleaned up.
-     Code Maintainability: Explicitly freeing memory makes the code easier to understand and maintain. It shows where and why resources are being allocated and deallocated.
-     Debugging: Explicitly freeing memory can help catch memory leaks earlier during program execution, making debugging easier.

Anyway, in this exact sample from the proposed changes, I can do remove free from the code. Please, confirm that this is the general practice for that repo.

Reply:
> Some comment to exit(3): While it is true that exit(3) ensures that all memory is freed, it is still considered good practice to explicitly free dynamically allocated memory using free() before calling exit(3) or before the program exits normally. This practice has several benefits:
> 
>     * ```
>       Portability: On some non-Linux systems, exit(3) might not guarantee that all resources are properly cleaned up.
>       ```
> 
>     * ```
>       Code Maintainability: Explicitly freeing memory makes the code easier to understand and maintain. It shows where and why resources are being allocated and deallocated.
>       ```

Having 13 different points where we `free(name);` a resource in the same function is exactly the opposite of code maintainability.  :)

> 
>     * ```
>       Debugging: Explicitly freeing memory can help catch memory leaks earlier during program execution, making debugging easier.
>       ```
> 
> 
> Anyway, in this exact sample from the proposed changes, I can do remove free from the code. Please, confirm that this is the general practice for that repo.

Reply:
I removed all free calls.